### DISCUSSION TITLE
What is the purpose behind the config merging in Laravel 11?

### DISCUSSION THREAD

**Discussion by ollieread** [laravel/framework#52958] [General]
The merging of the application config with the default config of the framework has caused an issue recently, by removing the ability to use numeric strings for some config values. By default, Laravel will merge in the following config values with the defaults:

 - `auth.guards`
 - `auth.providers`
 - `auth.passwords`
 - `broadcasting.connections`
 - `cache.stores`
 - `database.connections`
 - `filesystems.disks`
 - `logging.channels`
 - `mail.mailers`
 - `queue.connections`

However, from what I can tell, the default values within here are identical to the values provided with every Laravel install, which means if someone deletes a default connection from somewhere, it's added back.

My question is this, why? The only reason I can see behind this is that somewhere there are hardcoded references to default connections, though I doubt that's the case.

Does anyone know?

**Comment by islamsamy214** [comment]
Why do you want to remove them?

**Reply by ollieread** [reply]
What do you mean?

**Reply by islamsamy214** [reply]
I understood from the last four lines that you want to remove some values from the default config files!
that's why I wanted to ask why you want to remove them when you can directly override them

**Reply by ollieread** [reply]
> I understood from the last four lines that you want to remove some values from the default config files! that's why I wanted to ask why you want to remove them when you can directly override them

Oh, I have the merging disabled on my projects, so it's not an issue. What I'm trying to get at, is that I've no idea why it's even a thing. 

I would delete connections that I do not need. Having the default connections added back doesn't make sense.

**Reply by islamsamy214** [reply]
I think it's okay, it just makes things easier and faster when needed

**Reply by ollieread** [reply]
> I think it's okay, it just makes things easier and faster when needed

By merging in the default config, it technically makes it slower, though not by a noticeable amount, as well as increasing the memory usage because it has to remember more config, again, not by a noticeable amount.

I don't know what it makes easier because the frequency in which I have 5 different types of connections with the exact default settings is 0. I'd wager that's the case for most people too.

**Reply by islamsamy214** [reply]
Could you give an example, of what I'm imagining here if we are talking about `database.connection`, so it will never use any of them unless u declared that in the default connection or used it manually

**Reply by ollieread** [reply]
> Could you give an example, of what I'm imagining here if we are talking about `database.connection`, so it will never use any of them unless u declared that in the default connection or used it manually

I'm not sure what you mean. Laravels default database config comes with a number of connections. If you delete all the ones you don't use, say, SQLite, PostgreSQL, SQLServer and whatever other ones, Laravel adds them back. I don't understand why.
